image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl

```diff
@@ -36,6 +36,23 @@
 {{ $rox := deepCopy $values }}
 {{ $_ := set $ "_rox" $rox }}
 
+{{/*
+  The below section checks if any preexisting k8s secrets have been provided to the helm chart. 
+  If so, it looks them up in the cluster and loads them into the _rox config. This allows us to 
+  support pre-existing CA/sensor/admissionControl/collector secrets. This means an operator can
+  pull the necessary TLS certs from the central service dashboard, load them into the k8s cluster
+  ahead of time, and then deploy the secured-cluster-services helm chart without needing to put
+  TLS secrets in a values file that could get committed into source control.
+    */}}
+
+{{ include "srox.getExistingTLSSecret" (list $rox.ca $.Release.Namespace "cert" )}}
```

Comment:
I am not sure about this approach. I mean, this is essentially doing a `cluster->local->cluster` roundtrip, or am I missing something here? Have you considered to instead implement this in a way that lets you *reference* existing secrets without fetching them and then attempting to recreate them when the rendered manifests get applied to the server?

Also, a caveat: when the Helm chart is rendered within our operator Helm's lookup function is deliberately disabled. This would also suggest to use secret references instead (and skip rendering of the client secrets).

Reply:
That's a good point, I was caught up trying to replicate the combination of the CA cert with the service TLS key/cert into a single secret as opposed to updating the templates to reference the existing secret. I'll try to get some time on my end to look into this change in the approach. Some of it is motivated by how we deploy software in airgapped environments and being able to manage secrets separately from the helm chart.